### PR TITLE
IS 17846 authentication schemes

### PR DESCRIPTION
**Pull request by parhambarazesh**

### CODE REVIEW

File: hub/documentation/connectors/patterns/authentication-schemes.rst [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+     - The client id as provided in the connector configuration.
+     - String
+   * - ``oauth_client_secret``
+     - The client secret as provided in the connector configuration.
+     - String
+   * - ``oauth_refresh_token``
+     - The token used to refresh the access token (obtained during onboarding).
+     - String
+
+* Example of connectors using OAuth2: Hubspot, Keap, Shopify, Sage, ZohoCRM, Wave, BusinessCentral, Asana
```
Comment (erik-leven) [inline hub/documentation/connectors/patterns/authentication-schemes.rst:44]: links?

File: hub/documentation/connectors/patterns/authentication-schemes.rst [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+   * - Property
+     - Description
+     - Type
+   * - ``api_key``
+     - The api key as provided in the connector configuration.
+     - String
+
+* Api key is generally less secure than OAuth2, as it is a simple key that can be easily compromised. It is recommended to use OAuth2 whenever possible (some connectors support both schemes).
+* Example of connectors using API key: PowerofficeGo, WooCommerce, Wix
```
Comment (erik-leven) [inline hub/documentation/connectors/patterns/authentication-schemes.rst:75]: links?

File: hub/documentation/connectors/patterns/authentication-schemes.rst [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+Tripletex authentication
+------------------------
+
+This is a custom style of authentication used by the Tripletex connector. It requires the following secrets:
+
+.. list-table::
+   :widths: 20, 30
+   :header-rows: 1
+
+   * - Property
```
Comment (erik-leven) [inline hub/documentation/connectors/patterns/authentication-schemes.rst:87]: hmm, isn't both of these only required for test accounts. Prod accounts only require one of them?

File: hub/documentation/connectors/patterns/authentication-schemes.rst [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+     - Description
+     - Type
+   * - ``jwt_access_token``
+     - The JWT token used to authenticate the client (obtained during onboarding).
+     - String
+   * - ``jwt_refresh_token``
+     - The token used to refresh the access token (obtained during onboarding).
+     - String
+
+* Example of connectors using JWT: webCRM
```
Comment (erik-leven) [inline hub/documentation/connectors/patterns/authentication-schemes.rst:177]: link?

File: hub/documentation/connectors/patterns/authentication-schemes.rst (hunk 1 above)
Comment (parhambarazesh) [reply hub/documentation/connectors/patterns/authentication-schemes.rst:44]: I already have one link to an example connector. This is to mention some external systems that use each specific auth scheme. Should I make links to each individual connector?

File: hub/documentation/connectors/patterns/authentication-schemes.rst (hunk 2 above)
Comment (parhambarazesh) [reply hub/documentation/connectors/patterns/authentication-schemes.rst:75]: same

File: hub/documentation/connectors/patterns/authentication-schemes.rst (hunk 4 above)
Comment (parhambarazesh) [reply hub/documentation/connectors/patterns/authentication-schemes.rst:177]: same

File: hub/documentation/connectors/patterns/authentication-schemes.rst (hunk 1 above)
Comment (erik-leven) [reply hub/documentation/connectors/patterns/authentication-schemes.rst:44]: Is there a reason why not? If anything things get more accessible with the links. Just my personal preference though

File: hub/documentation/connectors/patterns/authentication-schemes.rst (hunk 2 above)
Comment (erik-leven) [reply hub/documentation/connectors/patterns/authentication-schemes.rst:75]: same

**Review by erik-leven — APPROVED** [review]

**Review by hglad — APPROVED** [review]

**Pull request merged**